Tests/MongoSwiftTests/BSONValueTests.swift

```diff
@@ -23,77 +23,76 @@ final class BSONValueTests: MongoSwiftTestCase {
         // Bool
         checkTrueAndFalse(val: true, alternate: false)
         // String
         checkTrueAndFalse(val: "some", alternate: "not some")
         // RegularExpression
         checkTrueAndFalse(
-            val: RegularExpression(pattern: ".*", options: ""),
-            alternate: RegularExpression(pattern: ".+", options: "")
+                val: .regex(RegularExpression(pattern: ".*", options: "")),
+                alternate: .regex(RegularExpression(pattern: ".+", options: ""))
         )
         // Timestamp
-        checkTrueAndFalse(val: Timestamp(timestamp: 1, inc: 2), alternate: Timestamp(timestamp: 5, inc: 10))
+        checkTrueAndFalse(val: .timestamp(Timestamp(timestamp: 1, inc: 2)),
+                          alternate: .timestamp(Timestamp(timestamp: 5, inc: 10)))
         // Date
         checkTrueAndFalse(
-            val: Date(timeIntervalSinceReferenceDate: 5000),
-            alternate: Date(timeIntervalSinceReferenceDate: 5001)
+                val: .date(Date(timeIntervalSinceReferenceDate: 5000)),
+                alternate: .date(Date(timeIntervalSinceReferenceDate: 5001))
         )
         // MinKey & MaxKey
         expect(MinKey()).to(bsonEqual(MinKey()))
         expect(MaxKey()).to(bsonEqual(MaxKey()))
         // ObjectId
-        checkTrueAndFalse(val: ObjectId(), alternate: ObjectId())
+        checkTrueAndFalse(val: .objectId(ObjectId()), alternate: .objectId(ObjectId()))
         // CodeWithScope
         checkTrueAndFalse(
-            val: CodeWithScope(code: "console.log('foo');"),
-            alternate: CodeWithScope(code: "console.log(x);", scope: ["x": 2])
+                val: .codeWithScope(CodeWithScope(code: "console.log('foo');")),
+                alternate: .codeWithScope(CodeWithScope(code: "console.log(x);", scope: ["x": 2]))
         )
         // Binary
         checkTrueAndFalse(
-            val: try Binary(data: Data(base64Encoded: "c//SZESzTGmQ6OfR38A11A==")!, subtype: .uuid),
-            alternate: try Binary(data: Data(base64Encoded: "c//88KLnfdfefOfR33ddFA==")!, subtype: .uuid)
+                val: .binary(try Binary(data: Data(base64Encoded: "c//SZESzTGmQ6OfR38A11A==")!, subtype: .uuid)),
+                alternate: .binary(try Binary(data: Data(base64Encoded: "c//88KLnfdfefOfR33ddFA==")!, subtype: .uuid))
         )
-        // Document
-        checkTrueAndFalse(
-            val: [
-                "foo": 1.414,
-                "bar": "swift",
-                "nested": [ "a": 1, "b": "2" ] as Document
-                ] as Document,
-            alternate: [
-                "foo": 1.414,
-                "bar": "swift",
-                "nested": [ "a": 1, "b": "different" ] as Document
-                ] as Document
-        )
-        // Check that when an array contains non-BSONValues, we return false
-        let arr = [[String: Int]()]
-        expect(arr.bsonEquals(arr)).to(beFalse())
+        // TODO SWIFT-230: unskip
```

Comment:
```suggestion
        // TODO SWIFT-630: unskip
```

Reply:
going to that ticket did take me back to maybe the stupidest bug I ever introduced in the driver though 🙂

Reply:
oh wow, lol